docs/tutorial/t-reassign-partitions.md

```diff
@@ -0,0 +1,220 @@
+
+```json
+{
+  "broker": 3,
+  "logDirs": [
+    {
+      "partitions": [
+        {
+          "partition": "__KafkaCruiseControlModelTrainingSamples-10",
+          "size": 0,
+          "offsetLag": 0,
+          "isFuture": false
+        },
+        ...
+```
+
+### Removing old brokers
+
+Following on from the previous example of `Adding new brokers`, let's proceed to safely remove the new broker unit. Before scaling down the Juju application, we must make sure to carefully move any existing data on units about to be removed, to another unit, so as to preserve data availability.
+
+In practice, this means running a `rebalance` Juju action as seen above, **BEFORE** scaling down the appilcation. This ensures that data is moved, prior to the unit becoming unreachable and permanently losing the data on it.
+
+> **NOTE** - As partition data is replicated across a finite number of units based on the Kafka cluster's configured `replication.factor` property (usually == 3), it is imperative to remove only 1 broker at a time, so as to avoid losing all available replicas for a given partition
+
+To remove the most recent broker unit 3 from the previous example, re-run the action with a new parameter value of `--mode=remove`
+
+```shell
+juju run cruise-control/0 rebalance mode=remove dryrun=false brokerid=3 --wait=10m
+```
+
+This does not remove the unit, but moves the partitions from the current broker unit to other brokers within the cluster.
+
+Once the action has completed, we can verify that broker 3 no longer has any assigned partitions by doing the following
+
+```bash
+juju ssh --container kafka kafka-k8s/leader \
+    '/opt/kafka/bin/kafka-log-dirs.sh' \
+    '--describe' \
+    '--bootstrap-server localhost:9093' \
+    '--command-config /etc/kafka/client.properties' \
+    | tail -n +3 | jq -c '.brokers[] | select(.broker == 3)' | jq
+```
+
+This should result something similar to the result seen below, with broker 3 now having no partitions assigned, completing the safe reassignment of a broker's data to other brokers in the cluster
+
+```json
+{
+  "broker": 3,
+  "logDirs": [
+    {
+      "partitions": [],
+      "error": null,
+      "logDir": "/var/lib/kafka/data"
+    }
+  ]
+}
+```
+
+Now, it is safe to scale-in the cluster, removing broker 3
+
+```shell
+juju scale-application kafka-k8s 3
+```
+
+### Full cluster rebalancing
+
+If a cluster has been running in production for some time, partition allocations may be imbalanced. This can happen naturally as topic load varies, topics are reconfigured to have more/less partitions, or new topics are added/removed etc. As such, it is expected that periodically during the lifetime of a deployment, an administrator will be called upon to better distribute the partition allocation across the existing broker units as part of cluster maintenance.
```

Comment:
```suggestion
If a cluster has been running in production for some time, partition allocations may be imbalanced. This can happen naturally as topic load varies, topics are reconfigured to have more/less partitions, or new topics are added/removed etc. As such, it is expected that periodically during the lifetime of a deployment, an administrator will run a command to redistribute the partition allocation across the existing broker units as part of cluster maintenance.
```